src/containerapp/azext_containerapp/custom.py

```diff
@@ -4721,15 +4721,17 @@ def create_containerapps_from_compose(cmd,  # pylint: disable=R0914
     # Validate managed environment
     parsed_managed_env = parse_resource_id(managed_env)
     managed_env_name = parsed_managed_env['name']
-
-    logger.info(  # pylint: disable=W1203
-        f"Creating the Container Apps managed environment {managed_env_name} under {resource_group_name} in {location}.")
+    env_rg = resource_group_name
+    if parsed_managed_env.get('resource_group'):
+        env_rg = parsed_managed_env.get('resource_group')
 
     try:
         managed_environment = show_managed_environment(cmd=cmd,
                                                        name=managed_env_name,
-                                                       resource_group_name=resource_group_name)
+                                                       resource_group_name=env_rg)
     except CLIInternalError:  # pylint: disable=W0702
+        logger.info(  # pylint: disable=W1203
+            f"Creating the Container Apps managed environment {managed_env_name} under {env_rg} in {location}.")
```

Comment:
if you use env_rg here, you have to use env_rg for the following create_containerapps_compose_environment() for consistence

Reply:
I will use `env_rg` to create environment.